### PR TITLE
[MOB-1931] add auth functions

### PR DESCRIPTION
**Pull request by roninopf**

### CODE REVIEW

File: ts/Iterable.ts [hunk 1]
```diff
@@ -219,28 +219,54 @@ class Iterable {
...
+  static setUserIdWithToken(userId: string | undefined, token: string | undefined) {
+    console.log("setUserId: " + userId + " token: " + token)
+    RNIterableAPI.setUserIdWithToken(userId, token)
+  }
+
+  /**
+   * Get the user ID of the current user
+   */
   static getUserId(): Promise<string | undefined> {
     console.log("getUserId")
```
Comment (Ayyanchira) [inline ts/Iterable.ts:271]: Should this PR also include `updateEmail` changes to `Iterable.ts` ? just like `setEmail` and `setUserId`

File: ts/Iterable.ts [hunk 2]
```diff
@@ -219,28 +219,54 @@ class Iterable {
...
+    RNIterableAPI.setEmail(email)
   }
 
+  /**
+   * Set the user of the SDK by email address and set the authentication token
+   * @param email the email address of the user
+   * @param token the authentication token of the user
+   */
+  static setEmailWithToken(email: string | undefined, token: string | undefined) {
+    console.log("setEmail: " + email + " token: " + token)
```
Comment (tapashmajumder) [inline ts/Iterable.ts:236]:
```suggestion
    console.log("setEmail: " + email + ", token: " + token)
```

File: ts/Iterable.ts [hunk 3]
```diff
@@ -219,28 +219,54 @@ class Iterable {
...
+    RNIterableAPI.setUserId(userId)
   }
 
+  /**
+   * Set the user of the SDK by ID and set the authentication token
+   * @param userId the ID of the user
+   * @param token the authentication token of the user
+   */
+  static setUserIdWithToken(userId: string | undefined, token: string | undefined) {
+    console.log("setUserId: " + userId + " token: " + token)
```
Comment (tapashmajumder) [inline ts/Iterable.ts:263]:
```suggestion
    console.log("setUserId: " + userId + ", token: " + token)
```

File: ios/RNIterableAPI/ReactIterableAPI.swift [hunk 4]
```diff
@@ -72,6 +72,12 @@ class ReactIterableAPI: RCTEventEmitter {
         IterableAPI.email = email
     }
     
+    @objc(setEmail:withToken:)
+    func set(email: String?, token: String) {
```
Comment (tapashmajumder) [inline ios/RNIterableAPI/ReactIterableAPI.swift:76]: token should be String?

File: ts/Iterable.ts (hunk 1 above)
Comment (roninopf) [reply ts/Iterable.ts:271]: Yep! Forgot it.

File: android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java [hunk 5]
```diff
@@ -88,19 +88,19 @@ public void setEmail(@Nullable String email) {
...
     public void updateEmail(String email) {
-        IterableLogger.d(TAG, "Update Email: " + email);
+        IterableLogger.d(TAG, "updateEmail: " + email);
         IterableApi.getInstance().updateEmail(email);
     }
 
     @ReactMethod
     public void updateEmail(String email, @Nullable String token) {
-        IterableLogger.d(TAG, "updateEmail: " + email + " token: " + token);
+        IterableLogger.d(TAG, "updateEmail: " + email + ", token: " + token);
```
Comment (Ayyanchira) [inline android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java:103]: Woah! Totally missed the commas and question marks in PR. Could have broken the code 😅

**Pull request merged**